library-js/src/scala/collection/mutable/ArrayBuilder.scala

```diff
@@ -225,7 +225,7 @@ object ArrayBuilder {
    *  @tparam T     type of elements for the array builder, subtype of `AnyRef` with a `ClassTag` context bound.
    */
   @SerialVersionUID(3L)
-  final class ofRef[T <: AnyRef](implicit ct: ClassTag[T]) extends ArrayBuilder[T] {
+  final class ofRef[T <: AnyRef | Null](implicit ct: ClassTag[T]) extends ArrayBuilder[T] {
 
     protected var elems: Array[T] = _
```

Comment:
Either this `var elems` and the ones in the other subclasses should have `| Null`, or the `def elems` in the superclass should not have `| Null`. Right now, it's inconsistent and quite confusing.

According to

> * Add | Null to a type if the code relies on checking for null: pattern-matching on null, comparing with null, or returning null.
> * If null is used to signal “not initialized”, it’s often clearer to make the field’s type explicitly nullable.

The consistent thing should be to use `| Null`.
```suggestion
    protected var elems: Array[T] | Null = null
```